### ISSUE TITLE
Include() adds unnecessary ORDER BY

### ISSUE THREAD

**Issue by grietine** [dotnet/efcore#22369]
### **Issue:**
EF Core generates unnecessary ORDER BY clause when using Include(), which slows down the whole query significantly.

EF Core **LINQ** example:
```C#
await ParentDbSet.Where(x => x.Type == "MyType").Include(x => x.Child).ToListAsync(cancellationToken)
```

**Expected query:**
```sql
SELECT * FROM [Parent] AS [p]
LEFT JOIN [Child] AS [c] ON [p].[Id] = [c].[ParentId]
WHERE [p].[Type] = 'MyType'
```

**Actual query:**
```sql
SELECT * FROM [Parent] AS [p]
LEFT JOIN [Child] AS [c] ON [p].[Id] = [c].[ParentId]
WHERE [p].[Type] = 'MyType'
ORDER BY [p].[Id], [c].[Id]
```

### Further technical details
EF Core version: 3.1
Database provider: Microsoft.EntityFrameworkCore.SqlServer 3.1.7
Target framework: NET Core 3.1

**Comment by roji** [comment]
Duplicate of #19571

**Comment by roji** [comment]
tl;dr the ORDER BY is necessary for the way EF Core loads the related entities. We do have #19828 tracking removal of the last ORDER BY, which indeed isn't necessary.
